src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2122,8 +2114,58 @@
       return(character())
 
    }
-   
+
    discoveries <- new.env(parent = emptyenv())
+
+   # for R Markdown docs, scan the YAML header (requires the YAML package, a dependency of
+   # rmarkdown)
+   if (requireNamespace("yaml", quietly = TRUE))
+   {
+      # split document into sections based on the YAML header delimter
+      sections <- unlist(strsplit(contents, "---", fixed = TRUE))
+      if (length(sections) > 2)
+      {
+         front <- yaml::read_yaml(text = sections[[2]])
+
+         # start with an empty output
+         output <- NULL
+
+         if (!is.null(names(front$output)))
+         {
+            # if the output key has children, it will appear as a list name
+            # output:
+            #   pkg_name::out_fmt:
+            #     foo: bar
+            output <- names(front$output)[[1]]
+         }
+         else if (is.character(front$output))
+         {
+            # if the output key doesn't have children, it will appear as a plain character
+            #
+            # output: pkg_name::out_fmt
+            output <- front$output
+         }
+
+         # check for references to an R package in output format
+         if (!is.null(output))
+         {
+            format <- unlist(strsplit(output, "::"))
```

Comment:
Is it possible that some users might want to reference unexported functions here; e.g. pkg:::bar?

Reply:
That's a good question. It is ill-advised and extremely unlikely (and if they do then I think it's okay that autodiscovery doesn't work!)